lib/hcrypto/evp.c

```diff
@@ -952,7 +978,11 @@ EVP_CipherFinal_ex(EVP_CIPHER_CTX *ctx, void *out, int *outlen)
 	memset(ctx->buf + ctx->buf_len, 0, left);
 	ret = (*ctx->cipher->do_cipher)(ctx, out, ctx->buf, blocksize);
 	memset_s(ctx->buf, blocksize, 0, blocksize);
-	if (ret != 1)
+        /*
+         * OpenSSL's EVP_Cipher() returns the number of bytes on success, 0 on
+         * failure.
```

Comment:
ctx->cipher->do_cipher() cannot be set to EVP_Cipher().   EVP_Cipher() calls ctx->cipher->do_cipher().